File: programmers/etc/level3/단속카메라.py
def solution(routes):
    answer = 0
    routes.sort(key=lambda t: t[1])
    end = -30001
    for s, e in routes:
        if end < s:
            end = e
            answer += 1
    return answer


routes = [[-20,-15], [-14,-5], [-18,-13], [-5,-3]]
print(solution(routes)) # 2

File: Java/programmers/etc/level3/단속카메라.java
package CodeTest.Java.programmers.etc.level3;

import java.util.*;

public class 단속카메라 {
    public int solution(int[][] routes) {
        int answer = 0;
        Arrays.sort(routes, (o1,o2)->o1[1]-o2[1]);
        int end = -30001;
        
        for (int[] route : routes) {
            int s = route[0], e = route[1];
            if (end < s) {
                end = e;
                answer++;
            }
        }
        
        return answer;
    }
}
